src/google/protobuf/compiler/parser.cc

```diff
@@ -1373,7 +1373,7 @@ bool Parser::ParseOption(Message* options,
           value_location.AddPath(
               UninterpretedOption::kNegativeIntValueFieldNumber);
           uninterpreted_option->set_negative_int_value(
-              -static_cast<int64>(value));
+              static_cast<int64>(-value));
```

Comment:
I'm pretty sure some compilers are not going to happy about -uint64 and may end up as compile errors.